### PR TITLE
Logging: Fix some edge cases when identifying if the logged messages comes from external code or not.

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Discussion in #14806

### Summary of the issue:
When logging NVDA tries to determine if the logged message comes from its own code or from an add-on. In some cases it does so incorrectly.
### Case 1: Directory prepended to `sys.path` by an add-on
Some add-ons have to modify `sys.path` to import additional libraries. NVDA's `logHandler` relied on the fact that the first entry in path represents the location of NVDA's source code. While add-on developers should really clean-up after themselves i.e. modify path only for  as long as necessary to import additional libraries this is not something we can enforce. Initially observed by @CyrilleB79 for an add-on for ChatGPT.
### Case 2: Messages logged from add-ons installed in the default config for source copies
By default configuration folder for source copies is placed next to NVDA sources, i.e. in the `source` folder. When determining if the path belongs to NVDA or not we used to check if the given path starts with NVDA's source folder which was true for the plugins in  the user config even though the code was not a part of NVDA.
### Description of user facing changes
This should affect only developers who inspect the log.
### Description of development approach
When checking if the path is external or not we no longer rely on `sys.path` - rather the location of the `logHandler` is used to determine the true location of NVDA's source code. In addition if the path is beneath the current NVDA's config it is always marked as external.
### Testing strategy:
- With add-on for ChatGPT installed ensured that messages logged from NVDA core are not marked as external for both source and binary copies
- When running from sources ensured that messages logged from plugins installed in the default config folder are marked as coming from external code.
### Known issues with pull request:
None known
### Change log entries:
For Developers
- NVDA should more accurately determine if the logged message is coming from its own code

### Code Review Checklist:

- [X] Pull Request description:
  - description is up to date
  - change log entries
- [X] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [X] API is compatible with existing add-ons.
- [X] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [X] Security precautions taken.

### CODE REVIEW

File: source/logHandler.py [hunk 1]
```diff
@@ -31,22 +31,33 @@
...
 
 
 def isPathExternalToNVDA(path: str) -> bool:
 	""" Checks if the given path is external to NVDA (I.e. not pointing to built-in code). """
 	if(
 		path[0] != "<"
 		and os.path.isabs(path)
-		and not os.path.normpath(path).startswith(sys.path[0] + "\\")
+		and not os.path.normpath(path).startswith(_NVDA_CODE_PATH + "\\")
+		or(
```
Comment (seanbudd) [inline source/logHandler.py:47]: Lint issue:

```suggestion
		or (
```

File: source/logHandler.py [hunk 2]
```diff
@@ -31,22 +31,33 @@
...
+			and path.startswith(globalVars.appArgs.configPath)
+		)
 	):
 		# This module is external because:
 		# the code comes from a file (fn doesn't begin with "<");
-		# it has an absolute file path (code bundled in binary builds reports relative paths); and
-		# it is not part of NVDA's Python code (not beneath sys.path[0]).
+		# it has an absolute file path (code bundled in binary builds reports relative paths);
+		# it is not part of NVDA's Python code; and
+		# it is placed in NVDA's config, so belongs to an add-on or a plugin in the scratchpad.
```
Comment (seanbudd) [inline source/logHandler.py:56]: should this be "or" not "and"?

```suggestion
		# it has an absolute file path (code bundled in binary builds reports relative paths); and
		# it is not part of NVDA's Python code
		# (i.e. outside of NVDA directory or in NVDA's config, so belongs to an add-on or a plugin in the scratchpad).
```

**Comment by lukaszgo1** [comment]
@seanbudd  All done.

File: source/logHandler.py [hunk 3]
```diff
@@ -31,22 +31,34 @@
...
+			and path.startswith(globalVars.appArgs.configPath)
+		)
 	):
 		# This module is external because:
 		# the code comes from a file (fn doesn't begin with "<");
 		# it has an absolute file path (code bundled in binary builds reports relative paths); and
-		# it is not part of NVDA's Python code (not beneath sys.path[0]).
+		# it is not part of NVDA's Python code
+		# (i.e. outside of NVDA directory or in NVDA's config,
+		# so belongs to an add-on or a plugin in the scratchpad).
```
Comment (seanbudd) [inline source/logHandler.py:57]: fix grammer

```suggestion
		# so ir belongs to an add-on or a plugin in the scratchpad).
```

**Review by seanbudd — APPROVED** [review]

**Pull request merged**